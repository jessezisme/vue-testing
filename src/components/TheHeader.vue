<template>
    <header>
        <ul>
            <li><router-link to="{name: 'home'}">Home</router-link></li>
            <li>About</li>
        </ul>        
    </header>
</template>