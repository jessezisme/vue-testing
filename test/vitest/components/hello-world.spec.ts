import { mount } from "@vue/test-utils";
import {
  beforeEach,
  describe,
  expect,
  bench,
  expectTypeOf,
  it,
  vi,
} from "vitest";
import TheHeader from "@/components/HelloWorld.vue";

describe("TheHeader Component", () => {
  let wrapper = mount(TheHeader);

  beforeEach(() => {
    wrapper = mount(TheHeader);
  });

  it("Mounted", async () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("Contains links", async () => {
    const links = wrapper.findAll("a");
    expect(links.length).toBeGreaterThanOrEqual(1);
  });
});
