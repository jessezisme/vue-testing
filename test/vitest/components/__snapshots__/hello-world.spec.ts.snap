// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheHeader Component > Mounts properly > "FOOBAR" 1`] = `"FOOBAR"`;
